vmx/src/vm.rs

```diff
@@ -97,6 +104,15 @@ impl VM {
         })
     }
 
+    /// Registers a [`HostQuery`] with the given `name`.
+    pub fn register_host_query<Q>(&mut self, name: &'static str, query: Q)
```

Comment:
Are we sure we always want to accept only static strings as query names? Otherwise Cow could be used for str.